syntax = "proto3";

package mq.proto.lua.actor;

message LuaAddress {
	string mailbox = 1;
	optional string script = 2;
}

message LuaMessage {
	LuaAddress destination = 1;
	LuaAddress sender = 2;
	Variant data = 99;
}

message Table {
	map<string, Variant> entries = 1;
}

message ImVec2 {
	optional float x = 1;
	optional float y = 2;
}

message ImVec4 {
	optional float x = 1;
	optional float y = 2;
	optional float z = 3;
	optional float w = 4;
}

message Variant {
	oneof value {
		double number = 1;
		bool boolean = 5;
		string str = 8;
		Table table = 10;
		ImVec2 imvec2 = 11;
		ImVec4 imvec4 = 12;
	}
}

